scripts/bug_retriever.py

```diff
@@ -60,15 +60,22 @@ def retrieve_bugs(self):
 
         # Try to re-download inconsistent bugs, up to three times.
         for i in range(3):
-            bug_ids = set(bug_snapshot.get_inconsistencies())
-            if len(bug_ids) == 0:
+            # We look for inconsistencies in all bugs first, then, on following passes,
+            # we only look for inconsistencies in bugs that were found to be inconsistent in the first pass
+            if i == 0:
+                inconsistent_bugs = bug_snapshot.get_inconsistencies(bugzilla.get_bugs())
+            else:
+                inconsistent_bugs = bug_snapshot.get_inconsistencies(inconsistent_bugs)
+            inconsistent_bugs_ids = set(list(map(lambda bug: bug["id"], inconsistent_bugs)))
```

Comment:
No need to convert to a list first, you can build the set directly:
```Python
inconsistent_bug_ids = set(bug["id"] for bug in inconsistent_bugs)
```

Reply:
Ohk.